module/zfs/spa.c

```diff
@@ -1166,6 +1166,13 @@ spa_activate(spa_t *spa, int mode)
 	 */
 	spa->spa_zvol_taskq = taskq_create("z_zvol", 1, defclsyspri,
 	    1, INT_MAX, 0);
+
+	/*
+	 * The taskq to upgrade datasets in this pool. Currently used by
+	 * feature SPA_FEATURE_USEROBJ_ACCOUNTING.
+	 */
+	spa->spa_upgrade_taskq = taskq_create("z_upgrade", boot_ncpus,
+	    defclsyspri, 1, INT_MAX, 0);
```

Comment:
Let's create this taskq with the TASKQ_DYNAMIC flag.  This way up to `boot_ncpus` threads will be created only when needed, otherwise we'll just have one.  This wasn't done for the z_zvol taskq since we needed to ensure it always ran tasks serially.